10-JavaScript/variables-conditionals-iteration.md

```diff
@@ -0,0 +1,308 @@
+- Gain exposure to how to do some kinds of iteration in JavaScript with `for` and `while`
+
+## We Declare Variables, Then Assign (Then Maybe Re-Assign)
+
+We've dealt with variables in Ruby plenty! We've made them, assigned them a value using the assignment operator (`=`), and modified them.
+
+```ruby
+my_favorite_greeting = "Hello World!"
+```
+
+All of those concepts hold up in JavaScript. We will continue to declare variables, assign values to these variables, re-assign them.
+
+However, in JavaScript, it's worth it to know that we have two distinct steps when dealing with variables: **declaration** and **assignment.** There are formal definitions of _declaration_ we could get into, but for now, let's think of it this way:
+
+In JavaScript, when we create a new variable with a new name, we need to declare it first. When we declare it, we want to give it two things:
+1. A unique name for the variable
+1. A variable keyword that will determine some of its properties around scope and re-assignment
+
+This is an example of variable declaration:
+
+```javascript
+let myFavoriteGreeting;
+```
+
+Giving a variable a unique name is something we do all the time in Ruby. The second principle is a detail that we'll get into soon.
+
+To _use_ this declared variable, we can _assign_ a value to it using the assignment operator (`=`), just like in Ruby. We can continue to re-assign its value using the assignment operator.
+
+```javascript
+let myFavoriteGreeting; // <- declaration
+myFavoriteGreeting = 'Hello World'; // <- assignment
+```
+
+We only need to declare a variable once. We never need to re-declare it.
+
+### We Can Do Declaration and Assignment Together
+
+By the way, we can do declaration and initial assignment on one line:
+
+```javascript
+let clicheGoodbye = 'Don\'t forget to like and subscribe';
+```
+
+## Variables
+
+> What's the syntax for defining a variable in this language? What's the syntax for accessing them?
+
+> _-- Some cool programmer learning a new language_
+
+In JavaScript, in order to define a variable, we do explicit variable declaration.
+
+There are three keywords to use to declare a variable, all specifying different things:
+
+1. `var`
+1. `let`
+1. `const`
+
+Observe variable declaration and assignment below:
+
+```javascript
+var dees_favorite_color = 'green';
+let dees_favorite_number = 72;
+const dees_favorite_game = 'night in the woods';
+```
+
+### We Prefer `let` for Reassignable Variables
+
+`var` and `let` declare variables that can be re-assigned. `var` and `let` mean subtly different things, both having to do with [scope](https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var-to-declare-a-variable) which we'll get to later.  **A good rule of thumb and our best practice moving forward is to always use `let` for variables we expect to be re-assigned.**
+
+It may make sense to use `var` if you are working with an older version of the JavaScript interpreter.
```

Comment:
Maybe also mention that they'll see `var` in older code on the internet (like SO posts)